src/zope/globalrequest/__init__.py

```diff
@@ -20,4 +20,4 @@ def setRequest(request):
 
 def clearRequest():
     """ clear the stored request object """
-    setRequest(None)
+    setRequest(marker)
```

Comment:
A little fix while testing. It is supposed to return the marker if the value is unset in order to make the ``factory`` work, but marker is different from None. I suppose nobody uses the factory feature in real life, but as I detected it I changed it to behave correctly.